tests/Contrib/Unit/OTLPHttpSpanConverterTest.php

```diff
@@ -0,0 +1,247 @@
+
+        $converter = new SpanConverter();
+        $row = $converter->as_otlp_span($span);
+
+        $this->assertInstanceOf(V1\Span::class, $row);
+
+        $this->assertSame($span->getContext()->getSpanId(), bin2hex($row->getSpanId()));
+        $this->assertSame($span->getContext()->getTraceId(), bin2hex($row->getTraceId()));
+
+        $this->assertSame($span->getSpanName(), $row->getName());
+
+        // $this->assertIsInt($row['timestamp']);
+        // // timestamp should be in microseconds
+        // $this->assertGreaterThan(1e15, $row['timestamp']);
+
+        // $this->assertIsInt($row['duration']);
+        // $this->assertGreaterThan(0, $row['duration']);
+
+        // $this->assertCount(1, $row['tags']);
+        
+        // /** @var Attribute $attribute */
+        // $attribute = $span->getAttribute('service');
+        // $this->assertEquals($attribute->getValue(), $row['tags']['service']);
+
+        // $this->assertCount(1, $row['annotations']);
+        // [$annotation] = $row['annotations'];
+        // $this->assertEquals('validators.list', $annotation['value']);
+
+        // [$event] = \iterator_to_array($span->getEvents());
+        // $this->assertIsInt($annotation['timestamp']);
+
+        // // timestamp should be in microseconds
+        // $this->assertGreaterThan(1e15, $annotation['timestamp']);
+    }
+
+    /**
+     * @test
+     */
+    // public function durationShouldBeInMicroseconds()
+    // {
+    //     $span = new Span('duration.test', SpanContext::generate());
+
+    //     $row = (new SpanConverter('duration.test'))->as_otlp_span($span);
+
+    //     $this->assertEquals(
+    //         (int) (($span->getEnd() - $span->getStart()) / 1000),
+    //         $row['duration']
+    //     );
+    // }
+
+    /**
+     * @test
+     */
+    public function tagsAreCoercedCorrectlyToStrings()
+    {
+        $span = new Span('tags.test', SpanContext::generate());
+
+        $listOfStrings = ['string-1','string-2'];
+        $listOfNumbers = [1,2,3,3.1415,42];
+        $listOfBooleans = [true,true,false,true];
+        $listOfRandoms = [true,[1,2,3],false,'string-1',3.1415];
+
+        $span->setAttribute('string', 'string');
+        $span->setAttribute('integer-1', 1024);
+        $span->setAttribute('integer-2', 0);
+        $span->setAttribute('float', '1.2345');
+        $span->setAttribute('boolean-1', true);
+        $span->setAttribute('boolean-2', false);
+        $span->setAttribute('list-of-strings', $listOfStrings);
+        $span->setAttribute('list-of-numbers', $listOfNumbers);
+        $span->setAttribute('list-of-booleans', $listOfBooleans);
+        $span->setAttribute('list-of-random', $listOfRandoms);
+
+        $converter = new SpanConverter();
+        $tags = $converter->as_otlp_span($span)->getAttributes();
+
+        // // Check that we can convert all attributes to tags
+        $this->assertCount(10, $tags);
+
+        // // Tags destined for Otlp must be pairs of strings
```

Comment:
Are these tests commented out purposefully?

Reply:
Needs some spring cleaning. These were left overs from before this code conformed to OTLP. I'll give this a look and might just be able to delete a lot of it. Might need to re-do these tests.

Reply:
Been hacking, I'm fixing these up to work as they're meant to. String, int, float, bool all work. Testing the arrays are testing me...

Reply:
All sorted, just needed to step away from the code.